EIPS/eip-6122.md

```diff
@@ -0,0 +1,189 @@
+---
+eip: 6122
+title: Forkid checks based on timestamps
+author: Marius van der Wijden (@MariusVanDerWijden)
+description: Modifies the forkid checks to work with timestamps and block numbers
+discussions-to: https://ethereum-magicians.org/t/eip-6122-forkid-checks-based-on-timestamps/12130
+status: Draft
+type: Standards Track
+category: Networking
+created: 2022-12-13
+requires: 2124
+---
+
+## Abstract
+
+[EIP-2124](eip-2124.md) proposed a way of identifying nodes on the p2p network based on their chain configuration via the forkid parameter. 
+It allows nodes to cut incompatible nodes off quickly which makes the P2P network more reliable.
+After the merge forks are scheduled by blocktime instead of blocknumber. This EIP updates the Forkid calculation with blocktime.
+
+## Motivation
+
+While the execution layer scheduled forks by blocknumber, the consensus layer scheduled forks by slot number. The slot number is a time based measurement. In order to schedule forks at the same time on the consensus and execution layer, the execution layer is forced to also schedule forks by timestamp after the merge.
+
+The forkid calculation allows peers to quickly determine the configuration of peers and disconnect peers that are misconfigured or configured for other networks.
+
+## Specification
+
+Each node maintains the following values:
+
+- **`FORK_HASH`**: IEEE CRC32 checksum (`[4]byte`) of the genesis hash and fork blocks numbers or timestamps that already passed.
+  - The fork block numbers or timestamps are fed into the CRC32 checksum in ascending order.
+  - If multiple forks are applied at the same block or time, the block number or timestamp is checksummed only once.
+  - Block numbers are regarded as `uint64` integers, encoded in big endian format when checksumming.
+  - Block timestamps are regarded as `uint64` integers, encoded in big endian format when checksumming.
+  - If a chain is configured to start with a non-Frontier ruleset already in its genesis, that is NOT considered a fork.
+- **`FORK_NEXT`**: Block number or timestamp (`uint64`) of the next upcoming fork, or `0` if no next fork is known.
```

Comment:
How does one distinguish between block number and timestamps? Based on that the current timestamp is around 30 bits and we're not going to reach that many blocks?

Or there's no need to distinguish? If so, please leave a note about this.